api/bin/chainflip-ingress-egress-tracker/src/main.rs

```diff
@@ -1,28 +1,91 @@
+#![feature(async_fn_in_trait)]
+use async_trait::async_trait;
 use cf_chains::{
-	btc::BitcoinNetwork, evm::SchnorrVerificationComponents, AnyChain, Bitcoin, Chain, Ethereum,
-	Polkadot,
+	address::ToHumanreadableAddress, evm::SchnorrVerificationComponents, AnyChain, Bitcoin, Chain,
+	Ethereum, Polkadot,
 };
-use cf_primitives::{Asset, BroadcastId, ForeignChain};
+use cf_primitives::{Asset, BroadcastId, ForeignChain, NetworkEnvironment};
 use chainflip_engine::{
 	settings::{insert_command_line_option, CfSettings, HttpBasicAuthEndpoint, WsHttpEndpoints},
 	state_chain_observer::client::{
-		chain_api::ChainApi, storage_api::StorageApi, StateChainClient, STATE_CHAIN_CONNECTION,
+		chain_api::ChainApi, storage_api::StorageApi, STATE_CHAIN_CONNECTION,
 	},
 };
 use clap::Parser;
 use config::{Config, ConfigBuilder, ConfigError, Environment, Map, Source, Value};
 use futures::FutureExt;
-use jsonrpsee::{core::Error, server::ServerBuilder, RpcModule};
 use pallet_cf_broadcast::TransactionOutIdFor;
 use pallet_cf_ingress_egress::DepositWitness;
+use redis::{aio::MultiplexedConnection, AsyncCommands};
 use serde::{Deserialize, Serialize};
 use state_chain_runtime::PalletInstanceAlias;
-use std::{collections::HashMap, env, net::SocketAddr};
+use std::{collections::HashMap, env, ops::Deref};
 use tracing::log;
 use utilities::{rpc::NumberOrHex, task_scope};
 
 mod witnessing;
 
+#[async_trait]
+pub trait Store: Sync + Send + 'static {
+	type Output: Sync + Send + 'static;
+
+	async fn save_to_array<S: Storable>(&mut self, storable: &S) -> anyhow::Result<Self::Output>;
+	async fn save_singleton<S: Storable>(&mut self, storable: &S) -> anyhow::Result<Self::Output>;
+}
+
+#[derive(Clone)]
+struct RedisStore {
+	con: MultiplexedConnection,
+}
+
+impl RedisStore {
+	const REDIS_EXPIRY_IN_SECONDS: u64 = 3600;
```

Comment:
this might be a bit to short for bitcoin deposits - maybe we can use something like 6 hours to be safe?

Reply:
Every 10 seconds the values are refreshed anyway for the bitcoin mempool watching.